sipa/utils/__init__.py

```diff
@@ -82,6 +87,34 @@ def support_hotline_available():
         return False
 
 
+def make_link(text):
+    pat_url = re.compile(  r'''(?x)((http|https)://(\S+))''')
+
+    for url in re.findall(pat_url, text):
+       text = text.replace(url[0], '<a href="%(url)s">%(url)s</a>' % {"url" : url[0]})
+
+    return text
+
+def meetingcal():
+    url = "https://agdsn.de/cloud/remote.php/dav/public-calendars/bgiQmBstmfzRdMeH?export"
+
+    calendar = icalendar.Calendar.from_ical(requests.get(url).text)
```

Comment:
Do the request in a separate line and include proper error handling:

- Set a short timeout (e.g. 1 second); Test this by changing the URL to an invalid URL
- Only continue if status code is 200
- Catch errors if returned data is not in ical format; Test this by calling `from_ical()` with a hard-coded string

In any case, an unreachable or slow calendar should not slow down SIPA or make SIPA unusable.